### ISSUE TITLE
Buggy Labs & Tests

### ISSUE THREAD

**Issue by sitieajmal** [learn-co-curriculum/javascript-arrays#36]
Chapters "Hoisting" and "JavaScript Arrays" have buggy labs. Fixing just the first issue gets you to "Passed Local Tests" in the Learn web interface, even though the Learn IDE shows fails or errors.

Layman translation: You can basically cheat these tests and move on to the next chapter without really fixing anything.

Doesn't really help those who actually want to learn. Admins please fix.

**Comment by catespinosa** [comment]
these two functions are supposed to be different but they both passed with the same code:

```
function addElementToBeginningOfArray (array, element){
  array.unshift(element);
  return array;
}


function destructivelyAddElementToBeginningOfArray(array, element){
  array.unshift(element);
  return array;
}
```

**Comment by DakotaLMartinez** [comment]
@gj 

Pretty sure this one has already been fixed, by PR #43

**Comment by gj** [comment]
Closed with #46. Thanks @sitieajmal and @DakotaLMartinez !
